### PR TITLE
improve error info of field selector

### PR DESCRIPTION
**Pull request by CaoShuFeng**
/assign @liggitt 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60785)

**Comment by CaoShuFeng** [comment]
ping @liggitt for approve.

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -709,7 +709,7 @@ func (b *Builder) visitorResult() *Result {
 	}
 
 	if len(b.resources) != 0 {
-		return &Result{err: fmt.Errorf("resource(s) were provided, but no name, label selector, or --all flag specified")}
+		return &Result{err: fmt.Errorf("resource(s) were provided, but no name, label selector, field selector or --all flag specified")}
```
Comment (liggitt) [inline pkg/kubectl/resource/builder.go]: not all components that use the resource build provide a field selector option... is this message going to be confusing?

**Comment by CaoShuFeng** [comment]
Closed according to liggitt's suggestion.

**Pull request closed without merging**